pkg/transaction/transaction.go

```diff
@@ -284,34 +285,47 @@ func (t *transactionService) prepareTransaction(ctx context.Context, request *Tx
 		notice that gas price does not exceed 20 as defined by max fee.
 	*/
 
-	gasPrice := request.GasPrice
-	if gasPrice == nil {
-		gasPrice, err = t.backend.SuggestGasPrice(ctx)
-		if err != nil {
-			return nil, err
-		}
-	}
-
-	gasTipCap, err := t.backend.SuggestGasTipCap(ctx)
+	gasFeeCap, gasTipCap, err := t.suggestedFeeAndTip(ctx, request.GasPrice, tipBoostPercent)
```

Comment:
I like the new merged routine.  From my understanding of EIP-1559, you'll never need one without the other.